src/main/react-front-end/src/views/Transfer/EndpointBrowseComponent.js

```diff
@@ -380,6 +380,61 @@ export default class EndpointBrowseComponent extends Component {
 		});
 	}
 
+	sftpPathHandler(endpoint, path, id) {
+		var uri = endpoint.uri;
+		const {setLoading} = this.props;
+		setLoading(true);
+		uri = makeFileNameFromPath(uri, path, "");
+		console.log(endpoint)
+		listFiles(uri, endpoint, id[id.length-1], (data) =>{
+			setLoading(false);
+			let sortedfiles = this.filenameAscendingOrderSort(data.files);
+			setFilesWithPathListAndId(sortedfiles, path, id, endpoint);
+			this.setState({directoryPath: path, ids: id});
+		}, (error) =>{
+			if(error === "500"){
+				this._handleError("Login Failed. Re-directing to OAuth page");
```

Comment:
This error handling does not really make sense. In the event of an error listing we should stay at the current position, aka the last list operation done. So we would list the parent path effectively.